src/dbus/server/dbus_thread_object_rcp.cpp

```diff
@@ -1754,6 +1771,17 @@ void DBusThreadObjectRcp::LeaveNetworkHandler(DBusRequest &aRequest)
     });
 }
 
+void DBusThreadObjectRcp::TakeoverLeaderHandler(DBusRequest &aRequest)
+{
+    auto    threadHelper = mHost.GetThreadHelper();
+    otError error        = OT_ERROR_NONE;
+
+    SuccessOrExit(error = otThreadBecomeLeader(threadHelper->GetInstance()));
```

Comment:
From the API doc, this should only be called when the device is attached. Could you add a check before issuing this call?